src/tools/_tools.font-size.scss

```diff
@@ -17,35 +18,53 @@
+        // `--ms<n>-ratio` CSS custom properties.
+        $_font-size: calc(var(--ms#{$font-size}-ratio) * 1rem);
+        font-size: $_font-size $important;
 
-            @if ($line-height == auto) {
-                // Calculate the line-height by using the generated
-                // `--ms<n>-line-height` CSS custom properties at runtime.
-                $_line-height: var(--ms#{$ms}-line-height);
-                @if $line-height-modifier {
-                    $_line-height: calc(var(--ms#{$ms}-line-height) + #{$line-height-modifier * $FLUIDMS-GLOBAL-BASELINE});
-                }
-                line-height: $_line-height $important;
-            } @else {
-                @if (type-of($line-height) == number or $line-height == "inherit" or $line-height == "normal") {
-                    line-height: $line-height $important;
-                } @else if ($line-height != "none" and $line-height != false) {
-                    @error "`#{$line-height}` is not a valid value for `$line-height`.";
-                }
+        @if ($line-height == auto) {
+            // Calculate the line-height at runtime by using the generated
+            // `--ms<n>-line-height` CSS custom properties.
+            $_line-height: var(--ms#{$font-size}-line-height);
+            // If the line-height is modified, add/subtract that from the
+            // calculated line-height.
+            @if $line-height-modifier {
+                $_line-height: calc(var(--ms#{$font-size}-line-height) + #{$line-height-modifier * $FLUIDMS-GLOBAL-BASELINE});
+            }
+            line-height: $_line-height $important;
+        } @else {
+            @if (type-of($line-height) == number or $line-height == "inherit" or $line-height == "normal") {
+                line-height: $line-height $important;
+            } @else if ($line-height != "none" and $line-height != false) {
+                @error "`#{$line-height}` is not a valid value for `$line-height`.";
+            }
+        }
+    }
+    // If `$font-size` is anything but a unitless number.
+    @else {
+        // Just output the passed in font-size value.
+        font-size: $font-size $important;
+        @if ($line-height == auto) {
+            // If the font-size is custom, just output the
+            // `$FLUIDMS-GLOBAL-LINE-HEIGHT` as `em` value, so we can use `calc`
+            // with it when a line-height modifier is used.
+            $_line-height: $FLUIDMS-GLOBAL-LINE-HEIGHT * 1em;
+            // If the line-height is modified, add/subtract that from the
+            // calculated line-height.
+            @if $line-height-modifier {
+                $_line-height: calc(#{$_line-height} + #{$line-height-modifier * $FLUIDMS-GLOBAL-BASELINE});
+            }
+            line-height: $_line-height $important;
+        } @else {
+            @if (type-of($line-height) == number or $line-height == "inherit" or $line-height == "normal") {
```

Comment:
Conditions in l64 and l66 are duplicates of l39 and l41.

Consider creating a function instead of duplicated and hard to read conditions

Reply:
What I could do is using a mixin to extract the code duplication like this:

```scss
// Only execute when `$font-size` is a unitless number.
@if (type-of($font-size) == number and unitless($font-size)) {
    // Calculate the font-size at runtime by using the generated
    // `--ms<n>-ratio` CSS custom properties.
    $_font-size: calc(var(--ms#{$font-size}-ratio) * 1rem);
    font-size: $_font-size $important;

    @if ($line-height == auto) {
        // Calculate the line-height at runtime by using the generated
        // `--ms<n>-line-height` CSS custom properties.
        $_line-height: var(--ms#{$font-size}-line-height);
        // If the line-height is modified, add/subtract that from the
        // calculated line-height.
        @if $line-height-modifier {
            $_line-height: calc(var(--ms#{$font-size}-line-height) + #{$line-height-modifier * $FLUIDMS-GLOBAL-BASELINE});
        }
        line-height: $_line-height $important;
    } @else {
        @include _line-height-pass-through();
    }
}
// If `$font-size` is anything but a unitless number.
@else {
    // Just output the passed in font-size value.
    font-size: $font-size $important;
    @if ($line-height == auto) {
        // If the font-size is custom, just output the
        // `$FLUIDMS-GLOBAL-LINE-HEIGHT` as `em` value, so we can use `calc`
        // with it when a line-height modifier is used.
        $_line-height: $FLUIDMS-GLOBAL-LINE-HEIGHT * 1em;
        // If the line-height is modified, add/subtract that from the
        // calculated line-height.
        @if $line-height-modifier {
            $_line-height: calc(#{$_line-height} + #{$line-height-modifier * $FLUIDMS-GLOBAL-BASELINE});
        }
        line-height: $_line-height $important;
    } @else {
        @include _line-height-pass-through();
    }
}

@mixin _line-height-pass-through() {
    @if (type-of($line-height) == number or $line-height == "inherit" or $line-height == "normal") {
        line-height: $line-height $important;
    } @else if ($line-height != "none" and $line-height != false) {
        @error "`#{$line-height}` is not a valid value for `$line-height`.";
    }
}
```

But does that really make the code more readable? IMO, it doesn’t improve the readability.

Reply:
I was thinking more about a function, e.g.

```scss
@function validLineHeight($value) {
  @return (type-of($value) == number or $value == "inherit" or $value == "normal")
}
```

usage
```scss
@if ( validLineHeight($line-height) ) {
  ...
}
```

IMO such an approach would improve the readability for these complex/verbose conditions and the already complex system

Reply:
Ah, I see. Didn’t consider this. I agree, that does make it more readable!